pineappl_cli/src/plot.rs

```diff
@@ -489,46 +506,84 @@ impl Subcommand for Opts {
+            let enable_abs = !enable_int;
             // TODO: only enable if there are EW corrections
             let enable_rel_ewonoff = enable_abs;
-            let enable_abs_pdfs = if nconvs == 1 || bins == 1 { "# " } else { "" };
+            let enable_abs_pdfs = !(nconvs == 1 || bins == 1);
             let enable_ratio_pdf = enable_abs_pdfs;
             let enable_double_ratio_pdf = enable_abs_pdfs;
-            let enable_rel_pdfunc = if nconvs == 1 || bins == 1 || self.no_conv_fun_unc {
-                "# "
-            } else {
-                ""
-            };
+            let enable_rel_pdfunc = !(nconvs == 1 || bins == 1 || self.no_conv_fun_unc);
             let enable_rel_pdfpull = enable_rel_pdfunc;
 
-            print!(
-                include_str!("plot.py"),
-                enable_int = enable_int,
-                enable_abs = enable_abs,
-                enable_rel_ewonoff = enable_rel_ewonoff,
-                enable_abs_pdfs = enable_abs_pdfs,
-                enable_ratio_pdf = enable_ratio_pdf,
-                enable_double_ratio_pdf = enable_double_ratio_pdf,
-                enable_rel_pdfunc = enable_rel_pdfunc,
-                enable_rel_pdfpull = enable_rel_pdfpull,
+            let config = format!(
+                "title = r\"{title}\"
+xlabel = r\"{xlabel}\"
+ylabel = r\"{ylabel}\"
+xlog = {xlog}
+ylog = {ylog}
+scales = {scales}
+enable_int = {enable_int}
+enable_abs = {enable_abs}
+enable_rel_ewonoff = {enable_rel_ewonoff}
+enable_abs_pdfs = {enable_abs_pdfs}
+enable_ratio_pdf = {enable_ratio_pdf}
+enable_double_ratio_pdf = {enable_double_ratio_pdf}
+enable_rel_pdfunc = {enable_rel_pdfunc}
+enable_rel_pdfpull = {enable_rel_pdfpull}
+output = r\"{output}\"",
+                enable_int = map_bool(enable_int),
+                enable_abs = map_bool(enable_abs),
+                enable_rel_ewonoff = map_bool(enable_rel_ewonoff),
+                enable_abs_pdfs = map_bool(enable_abs_pdfs),
+                enable_ratio_pdf = map_bool(enable_ratio_pdf),
+                enable_double_ratio_pdf = map_bool(enable_double_ratio_pdf),
+                enable_rel_pdfunc = map_bool(enable_rel_pdfunc),
+                enable_rel_pdfpull = map_bool(enable_rel_pdfpull),
                 xlabel = xlabel,
                 ylabel = ylabel,
-                xlog = xlog,
-                ylog = ylog,
+                xlog = map_bool(xlog),
+                ylog = map_bool(ylog),
                 title = title,
                 scales = self.scales,
                 output = output.to_str().unwrap(),
-                data = data_string,
+            );
+
+            let data = format!(
+                "data = {data_string}
+metadata = {{
+    {metadata}
+}}",
+                data_string = data_string,
                 metadata = format_metadata(&vector),
             );
+            let template = include_str!("plot.py");
+            // UNWRAP: if there are no markers the template is wrong
```

Comment:
I'm almost sure this is not the most efficient way to do things, but it is the only one I managed to make running ...

Reply:
Well, it's not completely bad: you're distributing it within the binary.

It's true that in some other respects it may be better to install it somewhere on the file system, and recover it at runtime. But in this way you do not depend at all on the file system and anything else, and the price is increasing the binary size of something that is most likely negligible...
(it will be 10-20KB?)

Reply:
nono, I don't mean the idea - it is the best we can do given the constraint (I think) and I will try to implement my idea [from above](https://github.com/NNPDF/pineappl/pull/312#issuecomment-2363034277) to keep the generated file simple and self-contained.

I was more worried of the string manipulation I'm doing here ... I was fighting with `split` and other stuff which did not work